### DISCUSSION TITLE
Chunk deallocation issue

### DISCUSSION THREAD

**Discussion by jouborges** [paternostrox/Minecraft4Unity#3] [Q&A] [answered]
When moving around the world, chunks seem to generate reliably but chunk dealloacation seems to not follow it by the same pace. It seems to only deallocate every other generation update. I tried to find the part of the code responsible for this but wasn't able to. Is this behavior intended?

**Comment by paternostrox** [comment, marked as answer]
The code responsible for this is located in TerrainManager.cs. In the main game scene (VoxelWorld.unity) you can view TerrainManager exposed attributes, including "Chunk Spawn Size" and "Chunk Keep Size", they control how many chunks get spawned and kept in the world.

By default I set the keep size to be slightly bigger than spawn size for performance reasons, but if you want to change them be free to do so.
